packages/react-sdk/src/components/ImportWhiteboardDialog/ImportWhiteboardDialog.tsx

```diff
@@ -172,11 +252,120 @@ export function ImportWhiteboardDialog({
         <Button
           onClick={handleOnImport}
           variant="contained"
-          disabled={importedWhiteboard?.isError}
+          disabled={error || loading}
         >
           {t('boardBar.importWhiteboardDialog.success.import', 'Import')}
         </Button>
       </DialogActions>
     </Dialog>
   );
 }
+
+function readNWB(file: File): Promise<{
+  name: string;
+  isError: boolean;
+  data?: WhiteboardDocumentExport;
+}> {
+  const reader = new FileReader();
+
+  const logger = getLogger('SettingsMenu');
+
+  return new Promise((resolve) => {
+    reader.onabort = () => logger.warn('file reading was aborted');
+    reader.onerror = () => logger.warn('file reading has failed');
+    reader.onload = () => {
+      if (typeof reader.result !== 'string') {
+        return;
+      }
+
+      try {
+        const jsonData = JSON.parse(reader.result);
+
+        if (isValidWhiteboardExportDocument(jsonData)) {
+          resolve({
+            name: file.name,
+            isError: false,
+            data: jsonData,
+          });
+        } else {
+          resolve({
+            name: file.name,
+            isError: true,
+          });
+        }
+      } catch (ex) {
+        const logger = getLogger('SettingsMenu');
+        logger.error('Error while parsing the selected import file', ex);
+
+        resolve({
+          name: file.name,
+          isError: true,
+        });
+      }
+    };
+
+    reader.readAsText(file);
+  });
+}
+
+async function readPDF(file: File): Promise<WhiteboardDocumentExport> {
+  await initPDFJs();
+
+  const logger = getLogger('SettingsMenu');
+  logger.debug('Reading PDF file', file.name);
+
+  const pdf = await loadPDF(await file.arrayBuffer());
+  logger.debug('PDF loaded', pdf.numPages);
+
+  logger.debug('Rendering PDF to images');
+  const images = await renderPDFToImages(pdf);
+
+  logger.debug('PDF rendered to images', images.length);
+  logger.debug('Uploading images to the server');
```

Comment:
I cannot see an upload happening here. Is this debug statement correct or is some code missing?

Reply:
Left over when removing the code. Will fix